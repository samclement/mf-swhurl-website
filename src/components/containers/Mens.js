import React from 'react'
import Head from 'next/head'
import Header from '../blocks/Header'

function Mens(props) {
  const { gender } = props
  return (
    <div>
      <Head>
        <title>mf.swhurl.com | {gender}</title>
      </Head>
      <Header gender={gender} />
    </div>
  )
}

Mens.getInitialProps = async ({ req, query }) => {
  return { gender: 'mens' }
}

export default Mens
