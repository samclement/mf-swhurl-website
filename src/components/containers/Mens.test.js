/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import Mens from './Mens.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {} }
Router.router = mockedRouter

describe('With Snapshot Testing', () => {
  it('Mens page renders correctly', () => {
    const { asFragment } = render(<Mens />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('Mens getInitialProps', async () => {
    const res = await Mens.getInitialProps({})
    expect(res).toMatchSnapshot()
  })
})
