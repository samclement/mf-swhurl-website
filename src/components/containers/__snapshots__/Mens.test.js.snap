// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Mens getInitialProps 1`] = `
Object {
  "gender": "mens",
}
`;

exports[`With Snapshot Testing Mens page renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c1 {
  text-align: center;
  text-transform: uppercase;
  width: 33%;
}

.c3 {
  padding: 0 10px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000;
  font-weight: normal;
}

.c4 {
  border: 1px solid #000;
  margin: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  width: 100%;
}

.c6 {
  width: 100%;
}

<div>
    <nav>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <a
            class="c2"
            href="/mens"
          >
            Men
          </a>
        </div>
        <div
          class="c3"
        >
          <h1>
            <a
              class="c2"
              href="/"
            >
              MATCHESFASHION
            </a>
          </h1>
        </div>
        <div
          class="c1"
        >
          <a
            class="c2"
            href="/womens"
          >
            Women
          </a>
        </div>
      </div>
    </nav>
    <hr
      class="c4"
    />
    <div
      class="c5"
    >
      <a
        aria-label="SHOP MEN’S. Shop all men's items on Matchesfashion!"
        class="c6"
        data-gender="mens"
        href="/mens"
        title="SHOP MEN’S"
      >
        <img
          alt="SHOP MEN’S. Shop all men's items on Matchesfashion!"
          class="c7"
          src="//assets.matchesfashion.com/devTest/translated/2019/pre-home/launch/qa/published/img/hero/mens-desk.jpg?quality=60&"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
